morecantile/models.py

```diff
@@ -765,9 +771,8 @@ def custom(
 
     def matrix(self, zoom: int) -> TileMatrix:
```

Comment:
> why not use lru_cache here? 

I've read some article that mentioned that using lru cache for class method might result in memory leaks 🤷